### PR TITLE
Kava Boost

### PR DESCRIPTION
**Pull request by adamrturman**
**NOTE**

1. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
2. The protocol is usually listed within 24 hours of merging the PR
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama):
Kava Boost

##### Twitter Link:
https://twitter.com/kava_platform

##### Website Link:
https://app.kava.io/

##### Logo (High resolution, preferably in .svg and .png, for application on both white and black backgrounds. Will be shown with rounded borders):
```
<svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0_15278_92663)">
<path d="M16 32C24.8366 32 32 24.8366 32 16C32 7.16344 24.8366 0 16 0C7.16344 0 0 7.16344 0 16C0 24.8366 7.16344 32 16 32Z" fill="#EC1164"/>
<path d="M15.9788 27.6125L18.9038 19.74H12.9863L15.9788 27.6125Z" fill="white"/>
<path d="M15.9125 11.8912L18.895 19.74H24.1463L15.9088 4.38745L7.85376 19.74H12.995L15.9125 11.8912Z" fill="white"/>
</g>
<defs>
<clipPath id="clip0_15278_92663">
<rect width="32" height="32" fill="white"/>
</clipPath>
</defs>
</svg>
```

##### Current TVL:
- KAVA: $28.00 M
- bKAVA: $29.17 M (this value is not currently being returned when running the test) 
- USDC: $1.19 M

Total $58.36 M

##### Chain:
KAVA

##### Short Description (to be shown on DefiLlama):
Kava Boost is a storage protocol that receives incentives for simply supplying.

##### Category (full list at https://defillama.com/categories) *Please choose only one:
Yield

##### Oracle used (Chainlink/Band/API3/TWAP or any other that you are using):
Chainlink

##### methodology (what is being counted as tvl, how is tvl being calculated):
All assets supplied to Kava Boost.

### CODE REVIEW

File: projects/kavaBoost.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    const block = await sdk.api.util.lookupBlock(timestamp, {chain:'kava'})
+    url += `?height=${block.block}`
+  }
+
+  const response = await utils.fetchURL(url);
+
+  for (let coin of response.data.result) {
+    const tokenInfo = generic(coin.denom);
+    const tokenName = coin.denom === 'bkava' ? 'bkava' : tokenInfo[0];
+    balances[tokenName] = coin.amount / 10 ** tokenInfo[1];
```
Comment (adamrturman) [inline projects/kavaBoost.js:18]: bKAVA is being added to `balances` within the `for` loop

```
{ 
  'bkava: 25586386.019095, 
  'usd-coin': 1187281.342515, 
  'kava': 25000000 
}
```

but it isn't returned when running `node test projects/kavaBoost.js`

```
--- kava ---
KAVA                      28.00 M
USDC                      1.19 M
Total: 29.19 M 

--- tvl ---
KAVA                      28.00 M
USDC                      1.19 M
Total: 29.19 M 

------ TVL ------
kava                      29.19 M

total                    29.19 M 
```

**Comment by llamatester** [comment]
The adapter at projects/kavaBoost.js exports TVL: 
        
 
 
    kava                      29.63 M
    
    total                    29.63 M

**Comment by g1nt0ki** [comment]
Hi @adamrturman, thanks for the PRs but I am having some trouble. Can you answer the following questions:

  - bKava is bonded kava? we currently do not count staking of cosmos native tokens, i.e we dont count chain native staking example: cardano or ICP staking. Liquid staking is for non-chain companies offering synth tokens like mSOL/stETH etc
  - What is the difference between kava earn and kava boost, feel like it is duplicate

**Comment by adamrturman** [comment]
> Hi @adamrturman, thanks for the PRs but I am having some trouble. Can you answer the following questions:
> 
> * bKava is bonded kava? we currently do not count staking of cosmos native tokens, i.e we dont count chain native staking example: cardano or ICP staking. Liquid staking is for non-chain companies offering synth tokens like mSOL/stETH etc
> * What is the difference between kava earn and kava boost, feel like it is duplicate

1. bKAVA is a liquid staking derivative of KAVA like stETH. Users stake KAVA with a validator as they have been doing. The new functionality in our most release allows users to supply their staked KAVA to our new Liquid module which then creates a bKAVA derivative. bKAVA is then supplied to the Earn module. The Earn module then supplies bKAVA to Boost so the user can earn an additional yield on bKAVA in addition to the normal staking APY users earn on their staked KAVA.
2. The Earn module is a yield aggregator protocol. It determines where to supply a given asset to earn the best yield for that asset. For now, we offer the highest yield in our Boost protocol which boosts the APY that users earn on various assets. The reason we created the Earn module rather than directly supply to Boost is to give ourselves the option in future iterations where Earn could supply to a different protocol than just Boost such as Curve if Curve offered a higher APY than what Boost offers.

**Pull request merged**